BlazarData/src/test/java/com/hubspot/blazar/data/service/BranchServiceTest.java

```diff
@@ -20,9 +20,9 @@
   private BranchService branchService;
 
   @Test
-  public void testUpsertBasic() throws Exception {
-    long before = System.currentTimeMillis();
-    Thread.sleep(1000); // because the timestamp is in seconds, we need to ensure we wait at least one.
+  public void itInsertsABranchWhenNonePresent() throws Exception {
+    // Round down to nearest second because mysql is 1s precision
+    long before = Math.floorDiv(System.currentTimeMillis(), 1000L) * 1000;
```

Comment:
Because the updated timestamp is handled by mysql it is at 1s precision. if the before value is not an even second than the between compare further down will fail, because the value in mysql will be lower than the before due to the precision used.